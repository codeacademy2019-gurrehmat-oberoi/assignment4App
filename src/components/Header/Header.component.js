import {
  View, Text,
} from 'react-native';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Header.component.style';

class Header extends Component {
  render() {
    const { title } = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{title}</Text>
      </View>
    );
  }
}

Header.propTypes = {
  title: PropTypes.string,
};
Header.defaultProps = {
  title: 'Default Header Title',
};
export default Header;
