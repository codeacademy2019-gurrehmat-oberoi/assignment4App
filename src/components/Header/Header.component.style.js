import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    // flex: 1,
    margin: 10,
    borderWidth: 2,
    backgroundColor: 'blue',
    // alignItems: 'stretch',
  },
  title: {
    fontSize: 30,
    alignSelf: 'flex-start',
    color: 'white',
    // textAlign: 'center',
  },
});

export default styles;
