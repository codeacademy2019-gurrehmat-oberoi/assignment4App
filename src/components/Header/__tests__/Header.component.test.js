import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header.component';

describe('Form Name Card', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
