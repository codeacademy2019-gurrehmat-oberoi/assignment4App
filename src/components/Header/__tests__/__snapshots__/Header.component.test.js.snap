// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Name Card should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "blue",
      "borderWidth": 2,
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "white",
        "fontSize": 30,
      }
    }
  >
    Default Header Title
  </Text>
</View>
`;
